tests/metadata.rs

```diff
@@ -42,3 +42,46 @@ fn it_rejects_invalid_format_versions() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn it_prints_workspace_metadata() -> Result<()> {
+    let project = project()?
+        .file(
+            "Cargo.toml",
+            r#"[workspace]
+members = ["foo", "bar", "baz"]
+"#,
+        )?
+        .file(
+            "baz/Cargo.toml",
+            r#"[package]
+name = "baz"
+version = "0.1.0"
+edition = "2021"
+    
+[dependencies]
+"#,
+        )?
+        .file("baz/src/lib.rs", "")?
+        .build()?;
+
+    project
+        .cargo_component("new foo")
+        .assert()
+        .stderr(contains("Created component `foo` package"))
+        .success();
+
+    project
+        .cargo_component("new bar")
+        .assert()
+        .stderr(contains("Created component `bar` package"))
+        .success();
+
+    project
+        .cargo_component("metadata --format-version 1")
+        .assert()
+        .stdout(contains("foo-interface 0.1.0").and(contains("bar-interface 0.1.0")))
+        .success();
```

Comment:
baz metadata will also need to be printed for rust-analyzer to work. Best to add an assert for that here